bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/SanityTestCommand.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    public SanityTestCommand(SanityFlags flags) {
+        super(CliSpec.<SanityFlags>newBuilder().withFlags(flags).withName(NAME).withDescription(DESC).build());
+    }
+
+    /**
+     * Flags for sanity command.
+     */
+    @Accessors(fluent = true)
+    @Setter
+    public static class SanityFlags extends CliFlags{
+
+        @Parameter(names = {"-e", "--entries"}, description = "Total entries to be added for the test (default 10)")
+        private int entries = 10;
+
+        @Parameter(names = { "-t",
+            "--timeout" }, description = "Timeout for write/read operations in seconds (default 1)")
+        private int timeout = 1;
+
+    }
+
+    @Override
+    public boolean apply(ServerConfiguration conf, SanityFlags cmdFlags) {
+        try {
+            return handle(conf, cmdFlags);
+        } catch (Exception e) {
+            throw new UncheckedExecutionException(e.getMessage(), e);
+        }
+    }
+
+    private boolean handle(ServerConfiguration conf, SanityFlags cmdFlags) throws Exception {
+        ClientConfiguration clientConf = new ClientConfiguration();
+        clientConf.addConfiguration(conf);
+        clientConf.setEnsemblePlacementPolicy(LocalBookieEnsemblePlacementPolicy.class);
+        clientConf.setAddEntryTimeout(cmdFlags.timeout);
+        clientConf.setReadEntryTimeout(cmdFlags.timeout);
+
+        BookKeeper bk = new BookKeeper(clientConf);
+        LedgerHandle lh = null;
+        try {
+            lh = bk.createLedger(1, 1, BookKeeper.DigestType.MAC, new byte[0]);
+            LOG.info("Create ledger {}", lh.getId());
+
+            for (int i = 0; i < cmdFlags.entries; i++) {
+                String content = "entry-" + i;
+                lh.addEntry(content.getBytes(UTF_8));
+            }
+
+            LOG.info("Written {} entries in ledger {}", cmdFlags.entries, lh.getId());
+
+            // Reopen the ledger and read entries
+            lh = bk.openLedger(lh.getId(), BookKeeper.DigestType.MAC, new byte[0]);
+            if (lh.getLastAddConfirmed() != (cmdFlags.entries - 1)) {
+                throw new Exception("Invalid last entry found on ledger. expecting: " + (cmdFlags.entries - 1)
+                                        + " -- found: " + lh.getLastAddConfirmed());
+            }
+
+            Enumeration<LedgerEntry> entries = lh.readEntries(0, cmdFlags.entries - 1);
+            int i = 0;
+            while (entries.hasMoreElements()) {
+                LedgerEntry entry = entries.nextElement();
+                String actualMsg = new String(entry.getEntry(), UTF_8);
+                String expectedMsg = "entry-" + (i++);
+                if (!expectedMsg.equals(actualMsg)) {
+                    throw new Exception("Failed validation of received message - Expected: " + expectedMsg
+                                            + ", Actual: " + actualMsg);
+                }
+            }
+
+            LOG.info("Read {} entries from ledger {}", entries, lh.getId());
```

Comment:
entries should be i?
`LOG.info("Read {} entries from ledger {}", *i*, lh.getId());`

Reply:
yeah, fixed